source/extensions/filters/http/aws_request_signing/config.cc

```diff
@@ -51,27 +52,45 @@ AwsRequestSigningFilterFactory::createFilterFactoryFromProtoTyped(
     const AwsRequestSigningProtoConfig& config, const std::string& stats_prefix, DualInfo dual_info,
     Server::Configuration::ServerFactoryContext& server_context) {
 
+  std::string region;
+  region = config.region();
+
+  if (region.empty()) {
+    auto region_provider = std::make_shared<Extensions::Common::Aws::RegionProviderChain>();
+    absl::optional<std::string> regionopt = region_provider->getRegion();
+    if (!regionopt.has_value()) {
+      throw EnvoyException("Region string cannot be retrieved from configuration, environment or "
+                           "profile/config files.");
+    }
+    region = regionopt.value();
+  }
+
   auto credentials_provider =
       std::make_shared<Extensions::Common::Aws::DefaultCredentialsProviderChain>(
-          server_context.api(), makeOptRef(server_context), config.region(),
+          server_context.api(), makeOptRef(server_context), region,
           Extensions::Common::Aws::Utility::fetchMetadata);
   const auto matcher_config = Extensions::Common::Aws::AwsSigningHeaderExclusionVector(
       config.match_excluded_headers().begin(), config.match_excluded_headers().end());
 
   std::unique_ptr<Extensions::Common::Aws::Signer> signer;
 
   if (getSigningAlgorithm(config) == SigningAlgorithm::SIGV4A) {
+    if (config.region().empty()) {
+      throw EnvoyException("Region parameter does not contain a SigV4A region set.");
+    }
+    // We use config.region here as environment or file store is not a valid region location for
+    // AWS_SIGV4A
```

Comment:
why region set cannot be ready from Env or Config file?
Can you explicitly call this limitation from Envoy docs associated with Sigv4A signing? Also mention the same on proto that region is not option for SIGV4A signing.

Reply:
there is currently no defined sdk behaviour for region sets. so the safest option is not to treat an environment/config/credentials retrieved region as valid.
will update proto to make this more clear

Reply:
i've also clarified this more explicitly in all doc references.

Reply:
> there is currently no defined sdk behaviour for region sets. so the safest option is not to treat an environment/config/credentials retrieved region as valid. will update proto to make this more clear

I've now had confirmation that there is in fact defined behaviour - which is to use `AWS_SIGV4A_SIGNING_REGION_SET` or profile/credentials string `sigv4a_signing_region_set` . This is now the implementation in this PR, rather than exception if region not specific for SigV4A